cpp/mg_utility/mg_graph.hpp

```diff
@@ -0,0 +1,264 @@
+  ///
+  /// @return Vector of valid edges
+  std::vector<TEdge> ExistingEdges() const {
+    std::vector<TEdge> output;
+    output.reserve(edges_.size());
+    for (const auto &edge : edges_) {
+      if (edge.id == Graph::kDeletedEdgeId)
+        continue;
+      output.push_back(edge);
+    }
+    return output;
+  }
+
+  /// Checks if edge is valid.
+  ///
+  /// Edge is valid if is created and if is not deleted.
+  ///
+  /// @return true if edge is valid, otherwise returns false
+  bool IsEdgeValid(TSize edge_id) const {
+    if (edge_id < 0 || edge_id >= edges_.size())
+      return false;
+    if (edges_[edge_id].id == kDeletedEdgeId)
+      return false;
+    return true;
+  }
+
+  /// Removes edge from graph.
+  ///
+  /// Recommendation is to use this method only in the tests.
+  ///
+  /// @param[in] u node id of node on same edge
+  /// @param[in] v node id of node on same edge
+  void EraseEdge(TSize u, TSize v) {
+    if (u < 0 && u >= nodes_.size())
+      throw mg_exception::InvalidIDException();
+    if (v < 0 && v >= nodes_.size())
+      throw mg_exception::InvalidIDException();
+
+    auto it = nodes_to_edge_.find(std::minmax(u, v));
+    if (it == nodes_to_edge_.end())
+      return;
+    TSize edge_id = it->second;
+
+    for (auto it = adj_list_[u].begin(); it != adj_list_[u].end(); ++it) {
+      if (edges_[*it].to == v || edges_[*it].from == v) {
+        edges_[*it].id = Graph::kDeletedEdgeId;
+        adj_list_[u].erase(it);
+        break;
+      }
+    }
+    for (auto it = adj_list_[v].begin(); it != adj_list_[v].end(); ++it) {
+      if (edges_[*it].to == u || edges_[*it].from == u) {
+        edges_[*it].id = Graph::kDeletedEdgeId;
+        adj_list_[v].erase(it);
+        break;
+      }
+    }
+
+    for (auto it = neighbours_[u].begin(); it != neighbours_[u].end(); ++it) {
+      if (it->edge_id == edge_id) {
+        neighbours_[u].erase(it);
+        break;
+      }
+    }
+    for (auto it = neighbours_[v].begin(); it != neighbours_[v].end(); ++it) {
+      if (it->edge_id == edge_id) {
+        neighbours_[v].erase(it);
+        break;
+      }
+    }
+  }
+
+  ///
+  /// Returns the Memgraph database ID from graph view
+  ///
+  /// @param node_id view's inner ID
+  ///
+  TSize GetMemgraphNodeId(uint64_t node_id) {
+    if (inner_to_memgraph_id_.find(node_id) == inner_to_memgraph_id_.end()) {
+      throw mg_exception::InvalidIDException();
+    }
+    return inner_to_memgraph_id_[node_id];
+  }
+
+  /// Removes all edges and nodes from graph.
+  void Clear() {
+    adj_list_.clear();
+    nodes_.clear();
+    edges_.clear();
+    nodes_to_edge_.clear();
+    neighbours_.clear();
+
+    memgraph_to_inner_id_.clear();
+    inner_to_memgraph_id_.clear();
+  }
+
+private:
+  // Constant is used for marking deleted edges.
+  // If edge id is equal to constant, edge is deleted.
+  static const TSize kDeletedEdgeId = std::numeric_limits<TSize>::max();
```

Comment:
make this `static const inline` or `static constexpr` (prefer this) because of the ODR, (if you add this header into multiple translation units and try to link them you'll get annoying and hard to parse errors)